### PR TITLE
JDK-8276772: Refine javax.lang.model docs

### PR DESCRIPTION
**Pull request by jddarcy**
Pass over javax.lang.model docs for some minor cleanups and improvements.

Before pushing, I'll do any needed copyright updates and reflow paragraphs.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8276772](https://bugs.openjdk.java.net/browse/JDK-8276772): Refine javax.lang.model docs

### Reviewers
 * [Iris Clark](https://openjdk.java.net/census#iris) (@irisclark - **Reviewer**) ⚠️ Review applies to d455dce07f8b314ae4d488ab47f49fc04efe894c
 * [Vicente Romero](https://openjdk.java.net/census#vromero) (@vicente-romero-oracle - **Reviewer**) ⚠️ Review applies to 57b3982cd8b082604b642c0a6289f6e4ed2efc83

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6286/head:pull/6286` \
`$ git checkout pull/6286`

Update a local copy of the PR: \
`$ git checkout pull/6286` \
`$ git pull https://git.openjdk.java.net/jdk pull/6286/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6286`

View PR using the GUI difftool: \
`$ git pr show -t 6286`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6286.diff">https://git.openjdk.java.net/jdk/pull/6286.diff</a>

</details>

### CODE REVIEW

**Review by irisclark — APPROVED** [review]

File: src/java.compiler/share/classes/javax/lang/model/element/AnnotationMirror.java [hunk 1]
```diff
@@ -58,7 +58,8 @@
      * The order of the map matches the order in which the
      * values appear in the annotation's source.
      *
-     * <p>Note that an annotation mirror of a marker annotation type
+     * @apiNote
+     * An annotation mirror of a marker annotation type
      * will by definition have an empty map.
      *
      * <p>To fill in default values, use {@link
```
Comment (liach) [inline src/java.compiler/share/classes/javax/lang/model/element/AnnotationMirror.java:65]: Is this paragraph supposed to be part of the api notes as well?

Comment (jddarcy) [reply src/java.compiler/share/classes/javax/lang/model/element/AnnotationMirror.java:65]: Yes; those are both intended to be part of the apiNote and both render as such.

File: src/java.compiler/share/classes/javax/lang/model/SourceVersion.java [hunk 2]
```diff
@@ -63,6 +63,8 @@
      *  14: switch expressions (pattern matching and records in
      *      preview, text blocks in preview again)
      *  15: text blocks (records and pattern matching in preview again)
+     *  16: records and pattern matching
+     *  17: sealed classes, floating-point always strict
```
Comment (vicente-romero-oracle) [inline src/java.compiler/share/classes/javax/lang/model/SourceVersion.java:67]: shouldn't it be mentioned that sealed classes are in preview in 17?

Comment (jddarcy) [reply src/java.compiler/share/classes/javax/lang/model/SourceVersion.java:69]: Pushed updated history of preview features.

Comment (vicente-romero-oracle) [reply src/java.compiler/share/classes/javax/lang/model/SourceVersion.java:69]: sorry my mistake

**Review by vicente-romero-oracle — APPROVED** [review]
lgtm

**Pull request closed without merging**